libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -696,33 +697,68 @@ impl Greenlight {
         sync_state: &SyncState,
         htlc_list: Vec<Htlc>,
     ) -> NodeResult<(SyncState, Vec<Payment>)> {
-        let mut node_client = self.get_node_client().await?;
+        let mut node_client1 = self.get_node_client().await?;
+        let mut node_client2 = node_client1.clone();
+        let mut node_client3 = node_client1.clone();
+        let mut node_client4 = node_client1.clone();
         let mut new_sync_state = sync_state.clone();
 
-        // list invoices
-        let created_invoices = node_client
-            .list_invoices(cln::ListinvoicesRequest {
-                index: Some(ListinvoicesIndex::Created.into()),
-                start: Some(sync_state.list_invoices_index.created),
-                ..Default::default()
-            })
-            .await?
-            .into_inner();
+        let created_invoices_fut = node_client1.list_invoices(cln::ListinvoicesRequest {
+            index: Some(ListinvoicesIndex::Created.into()),
+            start: Some(sync_state.list_invoices_index.created),
+            ..Default::default()
+        });
+        let updated_invoices_fut = node_client2.list_invoices(cln::ListinvoicesRequest {
+            index: Some(ListinvoicesIndex::Updated.into()),
+            start: Some(sync_state.list_invoices_index.updated),
+            ..Default::default()
+        });
+        let created_send_pays_fut = node_client3.list_send_pays(cln::ListsendpaysRequest {
+            index: Some(ListsendpaysIndex::Created.into()),
+            start: Some(sync_state.send_pays_index.created),
+            ..Default::default()
+        });
+        let updated_send_pays_fut = node_client4.list_send_pays(cln::ListsendpaysRequest {
+            index: Some(ListsendpaysIndex::Updated.into()),
+            start: Some(sync_state.send_pays_index.updated),
+            ..Default::default()
+        });
+
+        let (
+            created_invoices_res,
+            updated_invoices_res,
+            created_send_pays_res,
+            updated_send_pays_res,
+        ) = join!(
+            created_invoices_fut,
+            updated_invoices_fut,
+            created_send_pays_fut,
+            updated_send_pays_fut,
+        );
+
+        let created_invoices = created_invoices_res?.into_inner();
         if let Some(last) = created_invoices.invoices.last() {
             new_sync_state.list_invoices_index.created = last.created_index()
         }
-        let updated_invoices = node_client
-            .list_invoices(cln::ListinvoicesRequest {
-                index: Some(ListinvoicesIndex::Updated.into()),
-                start: Some(sync_state.list_invoices_index.updated),
-                ..Default::default()
-            })
-            .await?
-            .into_inner();
+        let updated_invoices = updated_invoices_res?.into_inner();
         if let Some(last) = updated_invoices.invoices.last() {
             new_sync_state.list_invoices_index.updated = last.created_index()
```

Comment:
Looking at the CLN API for listinvoices there is also an `updated_index`. Just want to confirm `created_index` is correct.